shared/src/util/JSONParse.ts

```diff
@@ -0,0 +1,31 @@
+/** Parses the given array (or undefined it it's not an array) */
+export function arrayfromJSON(json: any): Array<any> | undefined {
+  return json && json instanceof Array ? json : undefined;
```

Comment:
`undefined instanceof Array` (or `null` or `0` or any falsey value) is `false`, so you just need: 

```suggestion
  return json instanceof Array ? json : undefined;
```